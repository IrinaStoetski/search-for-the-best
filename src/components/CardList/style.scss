.card-list {
  width: 70vw;
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-gap: 24px;
}
