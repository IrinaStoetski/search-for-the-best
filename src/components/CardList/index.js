// @flow
import React from 'react';
// Components
import EmptyState from '../EmptyState';
import Card from '../Card';
// Styles
import './style.scss';

type Props = {
  items: Array<Object>,
  itemsCount: number,
}
const CardList = ({
  items,
  itemsCount,
}: Props) => (
  <>
    {items && items.length !== 0 && (
    <div className="card-list">
      {items && items.map((item) => (
        <Card
          avatar={item.owner.avatar_url}
          name={item.name}
          description={item.description}
          link={item.html_url}
        />
      ))}
    </div>
    )}
    {!items && !itemsCount && <EmptyState title="Let's start! Type something!" /> }
    {itemsCount === 0 && <EmptyState title="Nothing was founded :( Try another query" />}
  </>
);

export default CardList;
