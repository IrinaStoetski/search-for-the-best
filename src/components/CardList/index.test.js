/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import CardList from './index';

test('renders correctly CardList component', () => {
  const tree = renderer
    .create(<CardList />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
