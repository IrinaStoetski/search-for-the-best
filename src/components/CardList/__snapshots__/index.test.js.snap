// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly CardList component 1`] = `
<div
  className="empty-state"
>
  <svg
    height="158"
    width="255"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
    >
      <rect
        fill="#F9F9F9"
        height="135"
        rx="10"
        width="116"
        x="115"
        y="15"
      />
      <rect
        fill="#FFF"
        height="134"
        rx="10"
        width="116"
        x="120"
        y="10"
      />
      <path
        d="M130 12a8 8 0 00-8 8v114a8 8 0 008 8h96a8 8 0 008-8V20a8 8 0 00-8-8h-96zm0-4h96c6.627 0 12 5.373 12 12v114c0 6.627-5.373 12-12 12h-96c-6.627 0-12-5.373-12-12V20c0-6.627 5.373-12 12-12z"
        fill="#EEE"
      />
      <path
        d="M140 33h12a2 2 0 010 4h-12a2 2 0 010-4zm32 12h12a2 2 0 010 4h-12a2 2 0 010-4z"
        fill="#E1DBF1"
      />
      <path
        d="M184 33h12a2 2 0 010 4h-12a2 2 0 010-4z"
        fill="#EEE"
      />
      <path
        d="M152 57h12a2 2 0 010 4h-12a2 2 0 010-4z"
        fill="#E1DBF1"
      />
      <path
        d="M160 33h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FEE1D3"
      />
      <path
        d="M172 69h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FC6D26"
      />
      <path
        d="M192 45h4a2 2 0 010 4h-4a2 2 0 010-4zm12-12h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FEF0E8"
      />
      <path
        d="M156 81h8a2 2 0 010 4h-8a2 2 0 010-4z"
        fill="#E1DBF1"
      />
      <path
        d="M184 69h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FC6D26"
      />
      <path
        d="M204.5 65a2.5 2.5 0 110 5 2.5 2.5 0 010-5zm10 29a2.5 2.5 0 110 5 2.5 2.5 0 010-5z"
        fill="#E1DBF1"
      />
      <path
        d="M223.5 94a2.5 2.5 0 110 5 2.5 2.5 0 010-5zM172 33h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FC6D26"
      />
      <path
        d="M140 69h24a2 2 0 010 4h-24a2 2 0 010-4zm50.429 10h29.142c1.342 0 2.429 1.12 2.429 2.5s-1.087 2.5-2.429 2.5H190.43c-1.342 0-2.429-1.12-2.429-2.5s1.087-2.5 2.429-2.5h-.001z"
        fill="#EEE"
      />
      <path
        d="M175.429 94h29.142c1.342 0 2.429 1.12 2.429 2.5s-1.087 2.5-2.429 2.5H175.43c-1.342 0-2.429-1.12-2.429-2.5s1.087-2.5 2.429-2.5h-.001z"
        fill="#EFEDF8"
      />
      <path
        d="M140 45h24a2 2 0 010 4h-24a2 2 0 010-4z"
        fill="#6B4FBB"
      />
      <path
        d="M172 57h24a2 2 0 010 4h-24a2 2 0 010-4z"
        fill="#C3B8E3"
      />
      <path
        d="M140 57h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FEF0E8"
      />
      <path
        d="M172 81h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#6B4FBB"
      />
      <path
        d="M140 81h8a2 2 0 010 4h-8a2 2 0 010-4z"
        fill="#FC6D26"
      />
      <path
        d="M140 93h8a2 2 0 010 4h-8a2 2 0 010-4zm16 0h8a2 2 0 010 4h-8a2 2 0 010-4z"
        fill="#FEF0E8"
      />
      <path
        d="M140 105h12a2 2 0 010 4h-12a2 2 0 010-4z"
        fill="#EFEDF8"
      />
      <path
        d="M158.375 117h14.25c1.312 0 2.375.895 2.375 2s-1.063 2-2.375 2h-14.25c-1.312 0-2.375-.895-2.375-2s1.063-2 2.375-2z"
        fill="#FC6D26"
      />
      <path
        d="M190.375 108h14.25c1.312 0 2.375 1.12 2.375 2.5s-1.063 2.5-2.375 2.5h-14.25c-1.312 0-2.375-1.12-2.375-2.5s1.063-2.5 2.375-2.5z"
        fill="#6B4FBB"
      />
      <path
        d="M160 105h4a2 2 0 010 4h-4a2 2 0 010-4z"
        fill="#FEE1D3"
      />
      <path
        d="M178 117h4a2 2 0 010 4h-4a2 2 0 010-4zm36.5-9h5a2.5 2.5 0 110 5h-5a2.5 2.5 0 110-5z"
        fill="#FEF0E8"
      />
      <path
        d="M175.5 108h5a2.5 2.5 0 110 5h-5a2.5 2.5 0 110-5z"
        fill="#FC6D26"
      />
      <path
        d="M140 117h10a2 2 0 010 4h-10a2 2 0 010-4z"
        fill="#EEE"
      />
      <circle
        cx="43"
        cy="78"
        fill="#FFF"
        r="35"
      />
      <path
        d="M43 113C23.67 113 8 97.33 8 78s15.67-35 35-35 35 15.67 35 35-15.67 35-35 35zm0-4c17.12 0 31-13.88 31-31 0-17.12-13.88-31-31-31-17.12 0-31 13.88-31 31 0 17.12 13.88 31 31 31z"
        fill="#EEE"
      />
      <path
        d="M85 79c0-1.105.887-2 1.998-2h4.004c1.103 0 1.998.888 1.998 2 0 1.105-.887 2-1.998 2h-4.004A1.994 1.994 0 0185 79zm14 0c0-1.105.887-2 1.998-2h4.004c1.103 0 1.998.888 1.998 2 0 1.105-.887 2-1.998 2h-4.004A1.994 1.994 0 0199 79z"
        fill="#E5E5E5"
      />
      <path
        d="M219.869 125.403c-13.26 9.264-31.649 7.977-43.484-3.858-13.279-13.279-13.279-34.806 0-48.084 13.278-13.278 34.805-13.278 48.083 0 11.836 11.836 13.118 30.23 3.858 43.485.133.111.262.229.387.354l15.556 15.555a6.004 6.004 0 01-4.243 10.245 5.997 5.997 0 01-4.243-1.759l-15.555-15.556a6.051 6.051 0 01-.355-.387m-1.06-9.512c10.154-10.154 10.154-26.617 0-36.77-10.153-10.154-26.616-10.154-36.77 0-10.153 10.153-10.153 26.616 0 36.77 10.154 10.153 26.617 10.153 36.77 0"
        fill="#6B4FBB"
      />
      <path
        d="M43 97c-10.493 0-19-8.507-19-19s8.507-19 19-19 19 8.507 19 19-8.507 19-19 19m0-3c8.837 0 16-7.163 16-16s-7.163-16-16-16-16 7.163-16 16 7.163 16 16 16"
        fill="#D2CAEA"
      />
      <path
        d="M40 70a3.001 3.001 0 016 0v7.993a3.001 3.001 0 01-6 0V70m3 18.997a3 3 0 110-6 3 3 0 010 6"
        fill="#6B4FBB"
      />
    </g>
  </svg>
</div>
`;
